### ISSUE TITLE
User creation wrong param

### ISSUE THREAD

**Issue by the0neyouseek** [sqmk/huejay#63]
| App | Version |
|:----------------:|:-------:|
| Hue Api version | 1.14.0 |
| Huejay version   | 1.2.2 |

---

Hi,

I was trying to create a new user following the example but the response from Hue was always 

`Error: Philips Hue: 5, invalid/missing parameters in body`

So, looking into your code, I found that you do not use the `deviceType` attributes but instead the `name` one.

For it to work I had to change :
```js
user.deviceType = 'my_device_type'; // Default is 'huejay'
```
By :
```js
user.attributes.set("name","huejay");
```

**Comment by sqmk** [comment]
Name should never be configured directly as per API documentation.

When I run the following (using the example script in huejay), I don't have an issue creating a new user:

```javascript
#!/usr/bin/env node

'use strict';

let huejay = require('../../lib/Huejay');
let client = require('../init-client');

console.log('Attempting to create user.');
console.log('Make sure link button on bridge is pressed.');
console.log('...');

let user = new client.users.User;
user.deviceType = 'customDevice';

client.users.create(user)
  .then(user => {
    console.log(`New user created:`);
    console.log(`  Username: ${user.username}`);
    console.log(`  Device type: ${user.deviceType}`);
  })
  .catch(error => {
    if (error instanceof huejay.Error && error.type === 101) {
      return console.log(`Link button not pressed. Try again...`);
    }

    console.log(error.stack);
  });

```

**Comment by sqmk** [comment]
I also tested the example in the readme, and also not experiencing your issue:

```javascript
#!/usr/bin/env node

'use strict';

let huejay = require('../../lib/Huejay');
let client = require('../init-client');

console.log('Attempting to create user.');
console.log('Make sure link button on bridge is pressed.');
console.log('...');

let user = new client.users.User;

// Optionally configure a device type / agent on the user
user.deviceType = 'my_device_type'; // Default is 'huejay'

client.users.create(user)
  .then(user => {
    console.log(`New user created - Username: ${user.username}`);
  })
  .catch(error => {
    if (error instanceof huejay.Error && error.type === 101) {
      return console.log(`Link button not pressed. Try again...`);
    }

    console.log(error.stack);
  });

```

```
Attempting to create user.
Make sure link button on bridge is pressed.
...
New user created - Username: uEmZ8dAMlcEdvz3NDzfjP1dvvJx4pDYw284c6ayZ
```

```text
 Username: uEmZ8dAMlcEdvz3NDzfjP1dvvJx4pDYw284c6ayZ
 Device type: my_device_type
 Create date: 2016-08-19T23:48:29
 Last use date: 2016-08-19T23:48:29
```

**Comment by the0neyouseek** [comment]
I tried to use this code in a brand new [sails](https://github.com/balderdashy/sails) app Controller and it did not work. I'll update this thread when I get back from vacation with new infos.

**Comment by sqmk** [comment]
Feel free to reopen this (or create a new issue) if you happen to stumble upon the problem again.

**Comment by danieldooley** [comment]
I think it's worth noting that I've had this same issue.

My code looks like this:

```javascript
function createClient(bridge){

        var client = new huejay.Client({
            host: bridge.ip
        });

        console.log(client);
        var user = new client.users.User;
        console.log(user);

        log("Authenticating with Hue bridge. Please press the link button on the bridge.");
        client.users.create(user).then(function(user){
            /* Application Stuff */
        }).catch(function(err){
            if (err instanceof huejay.Error && err.type === 101) {
                error("Link button not pressed. Aborting.");
                throw(err);
            } else {
                error("Unknown Error - (createClient)");
                console.error(err);
                throw(err);
            }
        });

}
```

Output looks like this:

```
Client {
  config: 
   { host: '192.168.1.102',
     port: 80,
     username: undefined,
     timeout: 15000 },
  accessors: {} }
User {
  attributes: Attributes { attributes: { deviceType: 'huejay' }, changed: {} } }
HueControl: Authenticating with Hue bridge. Please press the link button on the bridge.
HueControl: Unknown Error - (createClient)
{ [Error: Philips Hue: 5, invalid/missing parameters in body] type: 5 }
```

The workaround that @the0neyouseek proposed worked for me. As far as I'm aware I'm on the latest version of the API and Huejay, with a gen 1 bridge.

**Comment by sqmk** [comment]
Hi @danieldooley , can you tell me the API version of your bridge? I'll test using your code once you confirm the version.

**Comment by sqmk** [comment]
No need to give me an API version. Looks like not explicitly providing a device type will cause this issue, even though a default one should be provided. My example in this issue, documentation, and examples in the code always explicitly set deviceType, and both of your use cases don't set it.

**Comment by sqmk** [comment]
I've released Huejay v1.4.1 which should fix this issue. You no longer have to specifically set ```user.deviceType```, or set internal attribute 'name', prior to creating the user. @danieldooley, let me know if your original code posted in this issue will work without any modifications.

Per docs, you should continue to set device type by way of ```user.deviceType``` if you don't want it to be "huejay".

**Comment by danieldooley** [comment]
Thanks @sqmk the code above works perfectly with version 1.4.1
